source/agora/test/ValidatorCleanRestart.d

```diff
@@ -93,16 +89,16 @@ version(none) unittest
     // Check if the validators in the set B have all the addresses for
     // current validators and previous validators except themselves.
     set_b.each!(node =>
-        retryFor(node.getNodeInfo().addresses.length ==
+        retryFor(node.getNodeInfo().addresses.length >=
             GenesisValidators + conf.outsider_validators,
-            5.seconds));
+            10.seconds));
 
     // Make all the validators of the set A disable to respond
-    set_a.each!(node => node.ctrl.sleep(6.seconds, true));
+    set_a.each!(node => node.ctrl.shutdown);
 
     // Block 21 with the new validators in the set B
-    network.generateBlocks(iota(GenesisValidators, cast(size_t) nodes.length),
-        Height(GenesisValidatorCycle + 1));
+    network.generateBlocks(iota(GenesisValidators, all_validators),
+        Height(GenesisValidatorCycle + 1), 20.seconds);
```

Comment:
That's why CI is not compiling, duration

Reply:
Thanks, wrong function for that param.